code/controllers/subsystem/mapping.dm

```diff
@@ -269,8 +269,14 @@ SUBSYSTEM_DEF(mapping)
 		if(isfile(file))
 			var/zlev = GLOB.space_manager.add_new_zlevel(AWAY_MISSION, linkage = UNAFFECTED, traits = list(AWAY_LEVEL,BLOCK_TELEPORT))
 			GLOB.space_manager.add_dirt(zlev)
-			GLOB.maploader.load_map(file, z_offset = zlev)
-			late_setup_level(block(locate(1, 1, zlev), locate(world.maxx, world.maxy, zlev)))
+			try
+				GLOB.maploader.load_map(file, z_offset = zlev)
+				late_setup_level(block(locate(1, 1, zlev), locate(world.maxx, world.maxy, zlev)))
+			catch(var/exception/e)
+				GLOB.space_manager.remove_dirt(zlev)
+				message_admins("Map template [name] threw an error while loading. Safe exit attempted, but check for errors.")
+				log_admin("Map template [name] threw an error while loading. Safe exit attempted.")
+				throw e
```

Comment:
This proc should **never** be advanced proc called anyway. This is unnecessary and will only encourage people to go "oh I can try and load gateways at runtime because theres safety now"

Reply:
This catches all errors that may happen, not just advanced proc call. See: Shuttle manipulator fails.

Reply:
Oh wait, this is the main map loader not the template loader

Reply:
Yeah no need for it here